packages/opds-web-client/src/utils/auth.ts

```diff
@@ -1,4 +1,44 @@
+import {
+  AuthProvider,
+  AuthMethod,
+  ClientSamlMethod,
+  ServerSamlMethod
+} from "./../interfaces";
+
+export const SAML_AUTH_TYPE = "http://librarysimplified.org/authtype/SAML-2.0";
+
 export function generateCredentials(username: string, password: string) {
   const btoaStr = btoa(`${username}:${password}`);
   return `Basic ${btoaStr}`;
 }
+
+export function flattenSamlProviders(providers: AuthProvider<AuthMethod>[]) {
+  return providers.reduce((flattened, provider) => {
+    if (isServerSamlProvider(provider)) {
+      return [...flattened, ...serverToClientSamlProviders(provider)];
+    }
+    return [...flattened, provider];
+  }, []);
+}
+
+function serverToClientSamlProviders(
+  provider: AuthProvider<ServerSamlMethod>
+): AuthProvider<ClientSamlMethod>[] {
+  return provider.method.links.map(idp => ({
+    method: {
+      href: idp.href,
+      type: provider.method.type,
+      description: getEnglishValue(idp.display_names) ?? "Unknown SAML Provider"
+    },
+    id: idp.href,
+    plugin: provider.plugin
+  }));
+}
+
+export const getEnglishValue = (arr: [{ language: string; value: string }]) =>
+  arr.find(item => item.language === "en")?.value;
```

Comment:
Eventually, this would be good to be something like `getLanguageValue` and the language "en" is passed in. But not needed right now. Do we imagine this to be a different language any time soon or ever?

Reply:
I asked Leonard this and he said that he can envision sometime in the future that they might internationalize. I can change it if you like but I don't think that's going to happen anytime soon.

Reply:
Yeah, it was just something to consider later on.